### DISCUSSION TITLE
Frontend error pages

### DISCUSSION THREAD

**Discussion by obulat** [WordPress/openverse#2612] [General]
I've listed some error page kinds that we should have in #2586, but I probably missed a lot.

What kind of error pages do we need to show on the frontend? What data should each error show?

## Kinds of errors

1. 404 for pages that do not exist (openverse.org/nonexistentpage) - show general message.
2. 404 for single media result pages with non-existent id - show the media type and the id.
3. 429 for throttled requests - when should the user retry, what was their allowed rate.
4. 500 for API internal server error - can be the same as no. 1? with an "Internal sever error" message.
5. Timeout errors

### Special cases

What should we show for the following client-side errors:

- For All content view: what should be done when only one of the media types receives an error?
- Errors on the "Load more" fetching?
- Should we display an error when an error is loading Related media, or hide the section?
- When the requests fail with 429, the user should see an error page with a message indicating what the user can do and when they can resume using Openverse.

@openverse-team, are there other error pages we should have? Should we display something else on the error pages?
What should we show in the special cases of client-side errors?

**Comment by AetherUnbound** [comment]
Thanks for thinking about all these cases Olga! I think in several of the special cases you've mentioned, maybe a pop-up might be warranted? For instance, the first 3 special cases you mention could all be pop-ups in my mind.

Don't we also have a 404 page(s) already?

**Comment by zackkrida** [comment]
Thanks so much for putting this together! Concerning the _pages_:

I think all of these could use the same template, but accept a "message" prop which allows for customizing the text. "This {mediaType} is no longer available. See more images of {query}" for single results, for example.

For the client-side UI errors:

I think a toast/pop up would be appropriate for these errors which result from a user interaction...throttling, load more failure, etc. For others, like the related images, I think we should show some kind of "related images failed to load" message inline in the section.

I can't think of other cases you haven't addressed, so nice work!
